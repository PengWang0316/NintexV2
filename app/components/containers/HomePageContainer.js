import React, { useEffect, Fragment } from 'react';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { withAuthenticator } from 'aws-amplify-react';

import { amplifyAuthSignOption } from '../../config';
import WorkflowRunInstanceChart from '../WorkflowRunInstanceChart';
import WorkflowLocationChart from '../WorkflowLocationChart';
import WorkflowCountCard from '../cards/WorkflowCountCard';
import InstanceCountCard from '../cards/InstanceCountCard';
import PublisherCountCard from '../cards/PublisherCountCard';
import HealthScoreCard from '../cards/HealthScoreCard';
import TopPublisherList from '../TopPublishersList';
import WorkflowActionUseWordCloud from '../WorkflowActionUseWordCloud';
import { currentAuthenticatedUser as currentAuthenticatedUserAction } from '../../actions/UserActions';

const styles = {
  cardContainer: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: 20,
  },
};

const HomePage = ({ user, currentAuthenticatedUser, classes }) => {
  useEffect(() => {
    if (!user) currentAuthenticatedUser();
  });

  return (
    <Fragment>
      <div className={classes.cardContainer}>
        <WorkflowCountCard />
        <InstanceCountCard />
        <PublisherCountCard />
        <HealthScoreCard />
      </div>
      <div className={classes.cardContainer}>
        <WorkflowRunInstanceChart />
        <WorkflowLocationChart />
      </div>
      <div className={classes.cardContainer}>
        <TopPublisherList />
        <WorkflowActionUseWordCloud />
      </div>
    </Fragment>

  );
};

HomePage.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  user: PropTypes.objectOf(PropTypes.any),
  currentAuthenticatedUser: PropTypes.func.isRequired,
};
HomePage.defaultProps = { user: null };

/* istanbul ignore next */
const mapStateToProps = state => ({ user: state.user });
/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
  currentAuthenticatedUser: user => dispatch(currentAuthenticatedUserAction(user)),
});
export default withAuthenticator(
  withStyles(styles)(connect(mapStateToProps, mapDispatchToProps)(HomePage)),
  amplifyAuthSignOption,
);
