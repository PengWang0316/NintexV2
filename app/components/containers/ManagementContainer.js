import React, { useEffect } from 'react';
import { withAuthenticator } from 'aws-amplify-react';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { amplifyAuthSignOption } from '../../config';
import { currentAuthenticatedUser as currentAuthenticatedUserAction } from '../../actions/UserActions';
import WorkflowTable from '../WorkflowTable';

const useStyles = makeStyles({
  rootDiv: {
    width: '95%',
    display: 'flex',
    justifyContent: 'center',
    padding: '30px 0 0 85px',
  },
});

const ManagementContainer = ({ user, currentAuthenticatedUser }) => {
  const classes = useStyles();
  useEffect(() => {
    if (!user) currentAuthenticatedUser();
  });
  return (
    <div className={classes.rootDiv}>
      <WorkflowTable />
    </div>
  );
};

ManagementContainer.propTypes = {
  user: PropTypes.objectOf(PropTypes.string),
  currentAuthenticatedUser: PropTypes.func.isRequired,
};
ManagementContainer.defaultProps = { user: null };

/* istanbul ignore next */
const mapStateToProps = state => ({ user: state.user });
/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
  currentAuthenticatedUser: user => dispatch(currentAuthenticatedUserAction(user)),
});

export default withAuthenticator(
  connect(mapStateToProps, mapDispatchToProps)(ManagementContainer),
  amplifyAuthSignOption,
);
