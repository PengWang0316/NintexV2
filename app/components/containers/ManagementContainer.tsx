import React, { useEffect, memo, Profiler } from 'react';
import { withAuthenticator } from 'aws-amplify-react';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';

import { amplifyAuthSignOption } from '../../config';
import { currentAuthenticatedUser as currentAuthenticatedUserAction } from '../../store/User/actions';
import WorkflowTable from '../WorkflowTable';
import { AppState } from '../../store/ConfigureStore';
import { UserType } from '../../store/User/types';
import onRenderCallback from '../../libs/ProfilerCallback';

interface Props {
  user: UserType | null;
  currentAuthenticatedUser: Function;
}

const useStyles = makeStyles({
  rootDiv: {
    width: '95%',
    display: 'flex',
    justifyContent: 'center',
    padding: '30px 0 0 85px',
  },
});

export const ManagementContainer = ({ user = null, currentAuthenticatedUser }: Props) => {
  const classes = useStyles({});
  useEffect(() => {
    if (!user) currentAuthenticatedUser();
  });
  return (
    <div className={classes.rootDiv}>
      <Profiler id="WorkflowTable" onRender={onRenderCallback}>
        <WorkflowTable />
      </Profiler>
    </div>
  );
};

/* istanbul ignore next */
const mapStateToProps = (state: AppState) => ({ user: state.user });
/* istanbul ignore next */
const mapDispatchToProps = { currentAuthenticatedUser: currentAuthenticatedUserAction };

export default withAuthenticator(
  connect(mapStateToProps, mapDispatchToProps)(memo(ManagementContainer)),
  amplifyAuthSignOption,
);
