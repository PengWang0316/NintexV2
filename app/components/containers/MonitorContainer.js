import React, { useEffect } from 'react';
import { withAuthenticator } from 'aws-amplify-react';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { amplifyAuthSignOption } from '../../config';
import MonitorList from '../MonitorList';
import { currentAuthenticatedUser as currentAuthenticatedUserAction } from '../../actions/UserActions';
import { fetchWorkflowsByUser as fetchWorkflowsByUserAction } from '../../actions/WorkflowActions';

const useStyles = makeStyles({
  rootDiv: {
    width: '95%',
    display: 'flex',
    justifyContent: 'center',
    padding: '30px 0 0 85px',
  },
});

let isFetching;

const MonitorContainer = ({
  user, currentAuthenticatedUser, workflows, fetchWorkflowsByUser,
}) => {
  const classes = useStyles();
  useEffect(() => {
    if (!user) currentAuthenticatedUser();
    if (!isFetching && !workflows.isFetched) {
      fetchWorkflowsByUser();
      isFetching = true;
    }
  });
  return (
    <div className={classes.rootDiv}>
      <MonitorList />
    </div>
  );
};

MonitorContainer.propTypes = {
  user: PropTypes.objectOf(PropTypes.string),
  workflows: PropTypes.objectOf(PropTypes.any).isRequired,
  currentAuthenticatedUser: PropTypes.func.isRequired,
  fetchWorkflowsByUser: PropTypes.func.isRequired,
};
MonitorContainer.defaultProps = { user: null };

/* istanbul ignore next */
const mapStateToProps = state => ({ ...state });
/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
  currentAuthenticatedUser: user => dispatch(currentAuthenticatedUserAction(user)),
  fetchWorkflowsByUser: () => dispatch(fetchWorkflowsByUserAction()),
});

export default withAuthenticator(
  connect(mapStateToProps, mapDispatchToProps)(MonitorContainer),
  amplifyAuthSignOption,
);
