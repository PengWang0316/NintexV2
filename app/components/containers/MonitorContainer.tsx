import React, { useEffect, memo, ReactElement } from 'react';
import { withAuthenticator } from 'aws-amplify-react';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';

import { amplifyAuthSignOption } from '../../config';
import MonitorList from '../MonitorList';
import { currentAuthenticatedUser as currentAuthenticatedUserAction } from '../../store/User/actions';
import { fetchWorkflowsByUser as fetchWorkflowsByUserAction } from '../../store/Workflows/actions';
import { AppState } from '../../store/ConfigureStore';
import { UserType } from '../../store/User/types';
import { Workflows } from '../../store/Workflows/types';

interface Props {
  user: UserType | null;
  workflows: Workflows;
  currentAuthenticatedUser: Function;
  fetchWorkflowsByUser: Function;
}

const useStyles = makeStyles({
  rootDiv: {
    width: '95%',
    display: 'flex',
    justifyContent: 'center',
    padding: '30px 0 0 85px',
  },
});

let isFetching = false;

export const MonitorContainer = ({
  user = null, currentAuthenticatedUser, workflows, fetchWorkflowsByUser,
}: Props): ReactElement => {
  const classes = useStyles({});
  useEffect(() => {
    if (!user) currentAuthenticatedUser();
    if (!isFetching && !workflows.isFetched) {
      fetchWorkflowsByUser();
      isFetching = true;
    }
  });
  return (
    <div className={classes.rootDiv}>
      <MonitorList />
    </div>
  );
};

/* istanbul ignore next */
const mapStateToProps = (state: AppState) => ({ ...state });
/* istanbul ignore next */
const mapDispatchToProps = {
  currentAuthenticatedUser: currentAuthenticatedUserAction,
  fetchWorkflowsByUser: fetchWorkflowsByUserAction,
};

export default withAuthenticator(
  connect(mapStateToProps, mapDispatchToProps)(memo(MonitorContainer)),
  amplifyAuthSignOption,
);
