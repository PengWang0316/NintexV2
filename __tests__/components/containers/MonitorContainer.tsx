import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import { MonitorContainer } from '../../../app/components/containers/MonitorContainer';
import JssProviderWrapper from '../../libs/JssProviderWrapper';

jest.mock('../../../app/components/MonitorList', () => () => <>MonitorList</>);

describe('MonitorContainer component', () => {
  const defaultProps = {
    user: null,
    workflows: {
      isFetched: false,
      data: {
        anc: {
          workflowId: 'string',
          workflowName: 'string',
          publishDate: 'string',
          publisher: 'string',
          tags: null,
        },
      },
    },
    currentAuthenticatedUser: jest.fn(),
    fetchWorkflowsByUser: jest.fn(),
  };
  beforeEach(() => jest.clearAllMocks());

  // test('');

  test('Snapshot', () => expect(renderer.create(JssProviderWrapper(<MonitorContainer {...{ ...defaultProps }} />)).toJSON()).toMatchSnapshot());
});
