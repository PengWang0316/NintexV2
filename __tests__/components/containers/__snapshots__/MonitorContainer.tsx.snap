// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonitorContainer component Snapshot 1`] = `
<div
  className="makeStyles-rootDiv-1"
>
  MonitorList
</div>
`;
